core/store/src/main/java/uk/gov/gchq/gaffer/store/Store.java

```diff
@@ -158,6 +159,18 @@ public void initialise(final Schema schema, final StoreProperties properties) th
         addExecutorService();
     }
 
+    public void initialise(final Schema schema, final StoreProperties properties, final SchemaOptimiser schemaOptimiser) throws StoreException {
```

Comment:
Instead of having to add this extra method and the extra optimiseSchema method, it would be neater to just override the createSchemaOptimiser() method in the AccumuloStore class:

```java
protected SchemaOptimiser createSchemaOptimiser() {
          return new SchemaOptimiser(new AccumuloSerialisationFactory());
}
```